### ISSUE TITLE
sticky header and help

### ISSUE THREAD

**Issue by limetech** [limetech/dynamix#154]
Something the pre-dynamix webGui had was a "sticky header" and a Help button that automatically turned off when you navigated to another page.

The idea behind the sticky header was to permit you to click that Help button no matter where you are scrolled in the page.  For a long page this is nice because currently you have to scoll to the top, click Help, then scroll back where you were.  A powerful feature I wanted in the Help system was that you could click it anytime easily to display the help text.

I think we should consider restoring this functionality.

**Comment by bergware** [comment]
My view is that new users will need help all the time, and want to see the information on every page they visit. It won't do good if help needs to be 'activated' each time the user changes to a different page. Burrying a help setting somewhere under settings - isn't intuitive, a new user won't be able to find it.

More seasonal users will use help on a on-demand basis, then again toggling it on/off shouldn't be a big issue for them.

Perhaps the on/off status visibility of help can be improved ?

**Comment by limetech** [comment]
I agree with your view to leave it activated until they turn it off.  Maybe longterm we introduce a concept of a 'locking' button.  Like on a hand drill or a gas handle - you know if you pull the trigger the drill runs or the gas flows and and if you release it stops.  But there is a 'locking' mechanism you can engage that locks the trigger.  Anyway this could be done with the button - if they click a certain corner of the button it 'locks'.

More important is the idea of bringing back sticky header.  For those who just need quick help on a certain input field it would permit them to see it without having to scroll up, click, scroll back down, scroll up, click again, scroll back down.

**Comment by bergware** [comment]
I haven't observed the "sticky-header" functionality before, will need to look in your original code (b10) to see how it works. Shouldn't be too hard to implement, I guess.

There were a couple of ideas raised on the forum too for improving the Help function, something for 6.1 perhaps ?

**Comment by limetech** [comment]
You know what a sticky header is right?  How about this: as the page scrolls up, the header also scrolls off the top of the window, but when then menu bar gets to the top it stays there.  That would be cool!

**Comment by bergware** [comment]
Ok, now it is clearer what you mean. I remember in the original version the header was fixed at the top and a scrolling area was made under it.

In Dynamix it is a *bit* different, and definitely not as easy to change as I thought ;)

**Comment by limetech** [comment]
http://www.webdesignerdepot.com/2014/05/how-to-create-an-animated-sticky-header-with-css3-and-jquery/

**Comment by bergware** [comment]
Yeah, but it isn't as easy as given on that site... (you want to try yourself)

**Comment by bergware** [comment]
Changing this would require restesting of all pages and functionality, sure you want to do this at this point or save it for the next version (will definitely give you more time) ?

**Comment by limetech** [comment]
Right this is not a big priority and best left for post 6.0 'stable'.
